sympy/stats/crv.py

```diff
@@ -531,11 +531,13 @@ def compute_density(self, expr, **kwargs):
 
         gs = solveset(expr - y, self.value, S.Reals)
 
-        if isinstance(gs, Intersection) and S.Reals in gs.args:
-            gs = list(gs.args[1])
+        
+        if isinstance(gs, Intersection):
+            if len(gs.args) == 2 and gs.args[0] is S.Reals:
+                        gs = gs.args[1]
```

Comment:
```suggestion
                gs = gs.args[1]
```

Reply:
Can the intersection ever be longer than 2? Will the Reals always be in slot 1? Perhaps better to raise an error if the conditions you handle are not met, e.g. `else: raise NotImplementedError('anticipating Intersection(Reals, Set)')`.